pkg/controller/fileintegrity/fileintegrity_controller.go

```diff
@@ -102,17 +239,43 @@ func (r *ReconcileFileIntegrity) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, err
 	}
 
-	// update the aide-conf configmap with the contents of the cr-provided one
+	// create the aide script configmap if it does not exist.
+	defaultAideScript := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: aideScriptConfigMapName, Namespace: fileIntegrityNamespace}, defaultAideScript)
+	if err != nil {
+		if !kerr.IsNotFound(err) {
+			reqLogger.Error(err, "error getting aide script")
+			return reconcile.Result{}, err
+		}
+		// does not exist, create it
+		createErr := r.client.Create(context.TODO(), defaultAIDEScript())
+		if createErr != nil {
+			reqLogger.Error(err, "error creating aide script")
+			return reconcile.Result{}, createErr
+		}
+	}
+
+	// handle configuration configmap
 	defaultAideConf := &corev1.ConfigMap{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: "aide-conf", Namespace: "openshift-file-integrity"}, defaultAideConf)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: defaultConfigMapName, Namespace: fileIntegrityNamespace}, defaultAideConf)
 	if err != nil {
-		reqLogger.Error(err, "error getting default aide config")
-		return reconcile.Result{}, err
+		if !kerr.IsNotFound(err) {
+			reqLogger.Error(err, "error getting default aide config")
+			return reconcile.Result{}, err
+		}
+		// does not exist, create it
+		createErr := r.client.Create(context.TODO(), defaultAIDEConfigMap())
+		if createErr != nil {
+			reqLogger.Error(err, "error creating default aide config")
+			return reconcile.Result{}, createErr
+		}
 	}
-	if _, ok := defaultAideConf.Data["aide.conf"]; !ok {
+	if _, ok := defaultAideConf.Data[defaultConfDataKey]; !ok {
```

Comment:
when would it be the case that it has no data? In that case we might want to delete the configMap entirely and trigger a reconcile; which would then create the default one.

Reply:
Following up on this one